src/Cli/dotnet/commands/dotnet-workload/install/IInstaller.cs

```diff
@@ -13,6 +13,10 @@ internal interface IInstaller : IWorkloadManifestInstaller
     {
         int ExitCode { get; }
 
+        string InstallWorkloadSet(string path);
```

Comment:
Can you explain why you think that's a better design? From what I see, that would basically mean downloading the nuget package (admittedly potentially from the cache, at least the second time) twice and would require duplicating a lot of logic around calculating manifest updates and executing them. As I have it, it downloads the workload set at the time it would otherwise be looking for advertising manifests. It's important that it installs the workloadset.json file _before_ it tries to calculate manifest updates because it needs that file to calculate manifest updates. And manifest update calculation has to happen before we can actually install the manifests.